torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -25,6 +25,9 @@ namespace distributed {
 namespace rpc {
 
 constexpr long kToMilliseconds = 1000;
+// This large time duration is for the timeoutMapCV_. We cannot use
+// std::chrono::time_point::max() due to a known overflow-related bug.
```

Comment:
Luca, I don't have the link, but we noticed in process_group changes that when std::chrono::time_point::max() got converted into the appropriate timespec struct for the underlying pthread_cond_timedwait() syscall, the timepoint value would overflow.

It could work if we used a smaller max value, or just use the non-timedwait version.

Reply:
Indeed in the processgroup agent we do either `wait` or `wait_until` depending on whether the timeout is infinite:
https://github.com/pytorch/pytorch/blob/15da26f8aafe50fb34d343d872c7461546d7a317/torch/csrc/distributed/rpc/process_group_agent.cpp#L783-L788

Could we at least explain the issue better in the comment rather than expecting the user to know it already?